OpenSim/Simulation/Model/Frame.cpp

```diff
@@ -181,6 +182,31 @@ void Frame::attachGeometry(OpenSim::Geometry* geom)
     updProperty_attached_geometry().adoptAndAppendValue(geom);
 }
 
+void Frame::scale(const SimTK::Vec3& scaleFactors)
+{
+    for (int i = 0; i < getProperty_attached_geometry().size(); ++i) {
+        Geometry& geo = upd_attached_geometry(i);
+        geo.upd_scale_factors() = geo.get_scale_factors()
+                                  .elementwiseMultiply(scaleFactors);
+    }
+}
```

Comment:
Ah interesting. Geometry is a Component, not a ModelComponent, so it doesn't have its own scale() function. What happens to Geometry that is not `attached_geometry`? It seems to not get scaled. If that's the case, it would be great to document this somewhere.

Reply:
> What happens to Geometry that is not `attached_geometry`?

Do you mean, for example, adding a Mesh to a Frame's `components` list property? Correct: that Mesh wouldn't get scaled (nor would meshes added to any other Component's `components` list). This PR does not change that behavior. There are a few strategies we could take: search for and scale Geometry residing in the `components` list, disallow Geometry in the `components` list, add documentation, address in a subsequent PR, etc. This issue isn't directly aligned with the purpose of this PR, but I can address it here if it isn't too involved. Perhaps @aseth1 has a preference.

Reply:
How about objects like ContactGeometry and WrapObjects, what happens to their Geometry?

Reply:
> How about objects like ContactGeometry and WrapObjects, what happens to their Geometry?

Same thing as before: WrapObjects get scaled (please see the changes to Wrap{Cylinder|Ellipsoid|Object|Sphere|Torus}.{h|cpp}) and [ContactGeometry throws](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Simulation/Model/ContactGeometry.cpp#L113). The purpose of this PR was not to fix every aspect of scaling for every component: I can't manage that in a single PR. If it is now important for contact geometry to be able to scale, I can work on that once this PR is merged and I've finished addressing #1973.

Reply:
I don't think this PR should try to fix scaling for previously-unscaled components. But it might be a good opportunity to quickly and simply document for users which components are not scaled, especially if this list is fresh in our minds right now.

Reply:
Added comments to `Frame::scale()` and ContactGeometry class.